### PR TITLE
[cloud][CLOUD-187] add unit tests for DB operations for email invitations

### PR DESCRIPTION
**Pull request by kopancek**
# Description
adding more unit tests

## Test plan
This change is just unit tests, so that should pretty much be it...

### CODE REVIEW

File: internal/database/org_invitations_test.go [hunk 1]
```diff
@@ -18,10 +18,10 @@ func TestOrgInvitations(t *testing.T) {
 		t.Skip()
 	}
 	t.Parallel()
-	db := dbtest.NewDB(t)
+	db := NewDB(dbtest.NewDB(t))
 	ctx := context.Background()
 
-	sender, err := Users(db).Create(ctx, NewUser{
+	sender, err := db.Users().Create(ctx, NewUser{
```
Comment (rafax) [inline internal/database/org_invitations_test.go:24]: For my information - is this a part of larger migration? Will we ever get rid of the previous mechanism? I think this difference (db.Users() vs Users(db)) caught me and @pjlast by surprise today, and I don't really know what's the differenece.

File: internal/database/org_invitations_test.go [hunk 2]
```diff
@@ -51,47 +52,73 @@ func TestOrgInvitations(t *testing.T) {
...
+				t.Fatal(err)
+			}
+		}
+		i, err = db.OrgInvitations().GetByID(ctx, i.ID)
+		if err != nil {
+			t.Fatal(err)
+		}
+		invitations = append(invitations, i)
 	}
+	oi1 := invitations[0]
```
Comment (rafax) [inline internal/database/org_invitations_test.go:113]: nit: I'm used to destructuring / unpacking collections on a single line, but up to you

File: internal/database/org_invitations_test.go (hunk 1 above)
Comment (unknwon) [reply internal/database/org_invitations_test.go:24]: I did not look into very deep on what made these two method behave differently, but from  call sites' perspective: we can't inject our [mocked DB handle](https://sourcegraph.com/search?q=context:global+repo:%5Egithub%5C.com/sourcegraph/sourcegraph%24+database.newmockdb&patternType=literal) with `Users(db)`, but `db.Users()` does.

**Review by pietrorosa77 — COMMENTED** [review]
lgtm

**Pull request merged**